### DISCUSSION TITLE
Which tool parser is better for deploying Qwen with vLLM?

### DISCUSSION THREAD

**Discussion by endNone** [QwenLM/Qwen3#1098] [General]
Which tool parser is better for deploying Qwen with vLLM?

**Comment by jklj077** [comment]
Please refer to our [documentation](https://qwen.readthedocs.io/en/latest/framework/function_call.html#vllm).

In short, for Qwen2.5, you should use "hermes" as "--tool-call-parser". vllm does not support the function calling for previous versions of Qwen.

**Comment by endNone** [comment]
Thank you very much！
